upgrade/upgrade-provider.go

```diff
@@ -190,19 +190,30 @@ func UpgradeProvider(ctx Context, repoOrg, repoName string) error {
 	if ctx.UpgradePfVersion {
 		discoverSteps = append(discoverSteps,
 			step.F("Planning Plugin Framework Update", func() (string, error) {
-				latest, err := latestRelease(ctx, "pulumi/pulumi-terraform-bridge")
+				refs, err := gitRefsOf(ctx, "https://"+modPathWithoutVersion(goMod.Bridge.Path),
+					"tags")
 				if err != nil {
 					return "", err
 				}
+				sortedRefs := refs.sortedLabels(func(a string, b string) bool { return a > b })
```

Comment:
I'm waiting for a pf based provider upgrade to test this with

Reply:
hi! you can test this with pulumi-cloudflare! there's a [patch upgrade of bridge/pf](https://github.com/pulumi/pulumi-terraform-bridge/releases/tag/pf%2Fv0.15.1) and cloudflare could use that patched through.

Reply:
I think it looks good now https://github.com/pulumi/pulumi-cloudflare/pull/462

Reply:
@guineveresaenger do you mind giving this another pass when possible?